src/lib/cdev/CMakeLists.txt

```diff
@@ -49,7 +49,11 @@ px4_add_library(cdev
 	CDev.hpp
 	${SRCS_PLATFORM}
 )
-target_compile_options(cdev PRIVATE ${MAX_CUSTOM_OPT_LEVEL})
+
+if (${PX4_PLATFORM} STREQUAL "nuttx")
+	target_compile_options(cdev PRIVATE ${MAX_CUSTOM_OPT_LEVEL} -D__KERNEL__)
```

Comment:
Should `-D__KERNEL__` be set more generally for all the modules kernel side?

Reply:
It is set for the modules in kernel side, but this doesn't propagate to the libraries from which they depend on. And in general, libraries can be linked to both sides, cdev was a special vase. But I will check this one again tomorrow, maybe it is not needed any more, the code has cleaned up a lot since I added it. Thanks for review!

Reply:
Adding the flag is useless, and I removed it. It was leftovers from some previous solution, where I included nuttx internal header iin the cdev code to be able to free the node name. When I moved both allocating and freeing the node name in uORBDeviceNode class, this is no longer needed. Thanks again for pointing this out